docs/DeveloperGuide.md

```diff
@@ -148,8 +148,8 @@ Here are the other classes in `Logic` (omitted from the class diagram above) tha
 How the parsing works:
 
 * When called upon to parse a user command, the `NutriGoalsParser` class creates an `XYZCommandParser` (`XYZ` is a
-  placeholder for the specific command name e.g., `AddCommandParser`) which uses the other classes shown above to parse
-  the user command and create a `XYZCommand` object (e.g., `AddCommand`) which the `NutriGoalsParser` returns back as
+  placeholder for the specified command name e.g., `AddCommandParser`). This new class then uses the other classes shown above to parse
+  the user command and create a `XYZCommand` object (e.g., `AddCommand`) which `NutriGoalsParser` then returns as
```

Comment:
What new class are you referring to here?

```suggestion
  placeholder for the specific command name e.g., `AddCommandParser`) which uses the other classes shown above to parse 
  the user command and create a `XYZCommand` object (e.g., `AddCommand`) which the `NutriGoalsParser` returns back as
```

Reply:
Suggestion might make the sentence too long and unreadable, better to split into smaller sentences.
The use of 'return back' is also grammatically incorrect.